### PR TITLE
Let CI run tests for supported matrix instructions

### PR DESCRIPTION
**Pull request by rosenrodt**
EDIT: rebased to latest `develop` branch 

Added tests for currently supported matrix instructions. This PR intends to get jenkins CI run matrix instructions so test cases are not meant to be comprehensive

- fp32 32x32x1x2
- fp32 32x32x2x1
- bf16 32x32x2x2
- fp16 32x32x4x2

Test result:
```
$ python3 -m pytest -v Tensile/Tests/pre_checkin/mfma/
================================================= test session starts =================================================
platform linux -- Python 3.5.2, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /mnt/Tensile, inifile: pytest.ini
collected 4 items                                                                                                     

Tensile/Tests/pre_checkin/mfma/test_mfma.py::test_sgemm_asm_mi32x32x2x1 PASSED                                  [ 25%]
Tensile/Tests/pre_checkin/mfma/test_mfma.py::test_hpa_bfloat16_gemm_asm_mi32x32x2x2 PASSED                      [ 50%]
Tensile/Tests/pre_checkin/mfma/test_mfma.py::test_sgemm_asm_mi32x32x1x2 PASSED                                  [ 75%]
Tensile/Tests/pre_checkin/mfma/test_mfma.py::test_hpa_hgemm_asm_mi32x32x4x2 PASSED                              [100%]

============================================ 4 passed in 150.43s (0:02:30) ============================================
```

### CODE REVIEW

**Comment by rosenrodt** [comment]
Any idea why all 3 mfma-related tests fail on CI? Our test machine shows otherwise—only fp32 mfma 32x32x1x2 fails [(full log here)](https://gist.github.com/rosenrodt/410c949ea72f8dd815c2d5e600e8d88e): 

```
======================================================== short test summary info =========================================================
PASSED Tensile/Tests/pre_checkin/mfma/test_mfma.py::test_hpa_bfloat16_gemm_asm_mi32x32x2x2
PASSED Tensile/Tests/pre_checkin/mfma/test_mfma.py::test_sgemm_asm_mi32x32x1x2
FAILED Tensile/Tests/pre_checkin/mfma/test_mfma.py::test_sgemm_asm_mi32x32x2x1 - SystemExit: 1
```

The test script is as follows
```shell
#!/bin/bash
cd Tensile/Tests
./create_tests.py pre_checkin/mfma
cd ../..
PYTHONPATH=. python3 -m pytest -rA Tensile/Tests/pre_checkin/mfma
```

**Comment by rosenrodt** [comment]
@AlexBrownAMD could you also provide a working config yaml for your recent PR #800? I am unable to repro a working kernel on my side, therefore it is now blocking the development of mfma fp16 kernel generators.

**Comment by AlexBrownAMD** [comment]
Here's a yaml file I was using for testing. I'll take a look at that other test you're seeing fail. 
[mfma_test.zip](https://github.com/ROCmSoftwarePlatform/Tensile/files/4232074/mfma_test.zip)

File: Tensile/Tests/pre_checkin/mfma/sgemm_asm_mi32x32x2x1.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      BenchmarkCommonParameters:
+        - KernelLanguage: ["Assembly"]
+        - EdgeType: ["ShiftPtr"]
+        - PrefetchLocalRead: [True]
+      ForkParameters:
+        - MatrixInstruction:
+          - [32, 32, 2, 1]
+        - PrefetchGlobalRead: [True]
+        - ThreadTile:
+          - [ 2, 32 ]
```
Comment (AlexBrownAMD) [inline Tensile/Tests/pre_checkin/mfma/sgemm_asm_mi32x32x2x1.yaml]: Thread tile 2x32 doesn't work yet, working on support for that right now

Comment (rosenrodt) [reply Tensile/Tests/pre_checkin/mfma/sgemm_asm_mi32x32x2x1.yaml]: it works. thanks

File: Tensile/Tests/yaml_only/test_config.py [hunk 2]
```diff
@@ -71,6 +71,9 @@ def configMarks(filepath, rootDir, availableArchs):
         ArchFail = "xfail-%s" % arch
         if markNamed(ArchFail) in marks:
             marks.append(pytest.mark.xfail)
+        ArchSkip = "skip-%s" % arch
+        if markNamed(ArchSkip) in marks:
+            marks.append(pytest.mark.skip)
```
Comment (AlexBrownAMD) [inline Tensile/Tests/yaml_only/test_config.py]: New marks should also be added to the list in pytest.ini

Comment (rosenrodt) [reply Tensile/Tests/yaml_only/test_config.py]: mark added to pytest.ini

**Comment by rosenrodt** [comment]
@sdquiring @AlexBrownAMD I ran the MFMA tests with `tox` and they look fine on my machine but not with our CI server. Did I miss anything? Here is the log below:

```
$ python3 -m tox -v --workdir /tmp/.tensile-tox -e py35 -vv -- ../Tensile/Tests -m 'mfma'
======================================================================== test session starts =========================================================================
platform linux -- Python 3.5.2, pytest-5.4.0, py-1.8.1, pluggy-0.13.1 -- /tmp/.tensile-tox/py35/bin/python
cachedir: /tmp/.tensile-tox/py35/.pytest_cache
rootdir: /Tensile, inifile: pytest.ini
plugins: xdist-1.31.0, forked-1.1.3
[gw0] linux Python 3.5.2 cwd: /Tensile
[gw1] linux Python 3.5.2 cwd: /Tensile
[gw2] linux Python 3.5.2 cwd: /Tensile
[gw3] linux Python 3.5.2 cwd: /Tensile
[gw0] Python 3.5.2 (default, Oct  8 2019, 13:06:37)  -- [GCC 5.4.0 20160609]
[gw1] Python 3.5.2 (default, Oct  8 2019, 13:06:37)  -- [GCC 5.4.0 20160609]
[gw2] Python 3.5.2 (default, Oct  8 2019, 13:06:37)  -- [GCC 5.4.0 20160609]
[gw3] Python 3.5.2 (default, Oct  8 2019, 13:06:37)  -- [GCC 5.4.0 20160609]
gw0 [4] / gw1 [4] / gw2 [4] / gw3 [4]
scheduling tests via LoadScheduling

Tensile/Tests/yaml_only/test_config.py::test_config[sgemm_asm_mi32x32x2x1]
Tensile/Tests/yaml_only/test_config.py::test_config[hpa_bfloat16_gemm_asm_mi32x32x2x2]
Tensile/Tests/yaml_only/test_config.py::test_config[hpa_hgemm_asm_mi32x32x4x2]
Tensile/Tests/yaml_only/test_config.py::test_config[sgemm_asm_mi32x32x1x2]
[gw0] [ 25%] PASSED Tensile/Tests/yaml_only/test_config.py::test_config[sgemm_asm_mi32x32x2x1]
[gw2] [ 50%] PASSED Tensile/Tests/yaml_only/test_config.py::test_config[hpa_hgemm_asm_mi32x32x4x2]
[gw1] [ 75%] PASSED Tensile/Tests/yaml_only/test_config.py::test_config[sgemm_asm_mi32x32x1x2]
[gw3] [100%] PASSED Tensile/Tests/yaml_only/test_config.py::test_config[hpa_bfloat16_gemm_asm_mi32x32x2x2]

========================================================================== warnings summary ==========================================================================
Tensile/SolutionStructs.py:2786
Tensile/SolutionStructs.py:2786
Tensile/SolutionStructs.py:2786
Tensile/SolutionStructs.py:2786
  /Tensile/Tensile/SolutionStructs.py:2786: UserWarning: Legacy client does not support MagicDivAlg==2, forcing MagicDivAlg=1
    warn("Legacy client does not support MagicDivAlg==2, forcing MagicDivAlg=1")

-- Docs: https://docs.pytest.org/en/latest/warnings.html
------------------------------------------------------------ generated xml file: /Tensile/py35_tests.xml -------------------------------------------------------------
============================================================= 4 passed, 4 warnings in 510.18s (0:08:30) ==============================================================
py35 finish: run-test  after 510.47 seconds
py35 start: run-test-post
py35 finish: run-test-post  after 0.00 seconds
______________________________________________________________________________ summary _______________________________________________________________________________
  py35: commands succeeded
  congratulations :)
cleanup /tmp/.tensile-tox/.tmp/package/1/Tensile-4.17.0.zip

```

**Pull request merged**